drivers/spi/spi_silabs_eusart.c

```diff
@@ -52,24 +91,26 @@ static int spi_silabs_eusart_configure(const struct device *dev, const struct sp
 
 	int err;
 
+	if (spi_context_configured(&data->ctx, config)) {
+		/* Runtime auto management disables and already configured.
+		 * No need to do it again, but must re-enable in case
+		 * TXEN/RXEN were cleared due to deep sleep.
+		 */
+#if !defined(CONFIG_PM_DEVICE_RUNTIME)
```

Comment:
This doesn't seem right. In runtime device PM mode, the device lifetime is decoupled from the system lifetime, meaning that this mode should be _more_ likely to need to re-enable the EUSART after deep sleep compared to system managed device PM.